mlflow/store/file_store.py

```diff
@@ -190,10 +190,11 @@ def list_experiments(self, view_type=ViewType.ACTIVE_ONLY):
         return experiments
 
     def _create_experiment_with_id(self, name, experiment_id, artifact_uri):
+        if not artifact_uri:
+            raise MlflowException(
+                "Experiment artifact URI must be a non-empty string, found: %s" % artifact_uri)
         self._check_root_dir()
         meta_dir = mkdir(self.root_directory, str(experiment_id))
-        artifact_uri = artifact_uri or path_to_local_file_uri(
-            os.path.join(self.root_directory, str(experiment_id)))
```

Comment:
Did we lose the logic which appends `experiment_id`? artifact_uri != root_artifact_uri. We should have a test to confirm this if this is a behavioral difference.

Reply:
Btw, it seems that the R tests caught this behavioral change, in `test-tracking-experiments.R` :)